src/main/java/oshi/software/os/mac/local/GlobalMemory.java

```diff
@@ -8,41 +8,68 @@
 package oshi.software.os.mac.local;
 
 import oshi.hardware.Memory;
-import oshi.util.ExecutingCommand;
+import oshi.software.os.mac.local.SystemB.VM_STATISTICS;
+import oshi.software.os.mac.local.SystemB.VM_STATISTICS64;
+
+import com.sun.jna.Platform;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.LongByReference;
 
 /**
- * Memory obtained by GlobalMemoryStatusEx.
+ * Memory obtained by host_statistics (vm_stat) and sysctl
  * 
- * @author alessandro[at]perucchi[dot]org
+ * @author widdis[at]gmail[dot]com
  */
 public class GlobalMemory implements Memory {
 
 	private long totalMemory = 0;
 
 	public long getAvailable() {
-		long returnCurrentUsageMemory = 0;
+		long availableMemory = 0;
 		long pageSize = 4096;
-		for (String line : ExecutingCommand.runNative("vm_stat")) {
-			if (line.matches("\\D+(page size of \\d+ bytes)\\D+")) {
-				pageSize = new Long(line.replaceAll("\\D", ""));
-			} else if (line.startsWith("Pages free:")
-					|| line.startsWith("Pages speculative:")
-					|| line.startsWith("Pages inactive:")) {
-				String[] memorySplit = line.split(":\\s+");
-				if (memorySplit.length > 1) {
-					returnCurrentUsageMemory += new Long(
-							memorySplit[1].replace(".", ""));
-				}
+
+		int machPort = SystemB.INSTANCE.mach_host_self();
+
+		LongByReference pPageSize = new LongByReference();
+		int ret = SystemB.INSTANCE.host_page_size(machPort, pPageSize);
+		if (ret == 0) {
+			pageSize = pPageSize.getValue();
+		}
+
+		if (Platform.is64Bit()) {
+			VM_STATISTICS64 vmStats = new VM_STATISTICS64();
+			ret = SystemB.INSTANCE.host_statistics64(machPort,
+					SystemB.HOST_VM_INFO64, vmStats, new IntByReference(
+							SystemB.HOST_VM_INFO64_COUNT));
+			if (ret == 0) {
+				availableMemory = (vmStats.free_count + vmStats.inactive_count)
+						* pageSize;
+				totalMemory = (vmStats.free_count + vmStats.inactive_count

```

Comment:
As you noted later, it sets the class variable for premature optimization.  Removing it would remove the need for the 64 bit VM_STATISTICS variant which will simplify oshi code (but I could leave the 64 bit version in the SystemB class with an eye towards later use in JNA library.